src/hotspot/share/oops/fieldStreams.hpp

```diff
@@ -248,63 +317,54 @@ class HierarchicalFieldStream : public StackObj  {
   }
 
   bool done() const { return _next_klass == nullptr && _current_stream.done(); }
+};
 
-  // bridge functions from FieldStreamBase
-  int index() const {
-    return _current_stream.index();
-  }
-
-  AccessFlags access_flags() const {
-    return _current_stream.access_flags();
-  }
-
-  FieldInfo::FieldFlags field_flags() const {
-    return _current_stream.field_flags();
-  }
-
-  Symbol* name() const {
-    return _current_stream.name();
-  }
-
-  Symbol* signature() const {
-    return _current_stream.signature();
-  }
-
-  Symbol* generic_signature() const {
-    return _current_stream.generic_signature();
-  }
-
-  int offset() const {
-    return _current_stream.offset();
-  }
-
-  bool is_contended() const {
-    return _current_stream.is_contended();
-  }
+/* Iterates on the fields of a class and its super-class top-down (java.lang.Object first)
+ * Doesn't traverse interfaces for now (let's decide how when/if the needs appear).
+ */
+template<typename FieldStreamType>
```

Comment:
By "doesn't traverse interfaces for now" I assume this traverses both static and non-static fields. Tbh I think traversing static fields seems questionable, or maybe I miss some cases, maybe this stream can discard those, too?